gitpod-io/gitpod
# Support ssh public keys configuration

## Description


This PR works on allowing users to upload their own ssh public key to Gitpod, make copy-paste ssh connect more conventional, and make sharing of ssh connect command more safety

[Link of Figma](https://www.figma.com/file/UTwRHTYhuF05IiaEWcL9MT/Upload-SSH-keys-to-Gitpod-%239932?node-id=1%3A2614)

## TODO
- [x] ‼️ Find out how to setup db-sync
- [x] ‼️ Merge PR https://github.com/gitpod-io/gitpod/pull/10617
- [x] ‼️ Deploy previous PR https://github.com/gitpod-io/gitpod/pull/10617
- [x] Dashboard ssh copy-paste improve
- [x] Make copy-paste ways can be switched
- [x] Properly content for users

~~Functions for late used time?~~ 

## Related Issue(s)

Relate https://github.com/gitpod-io/gitpod/issues/9932

## How to test


Modal of ssh copy-paste always shows `SSH Key` as the default selection, if the user has no public key uploaded, it will show a warning in this section.

- Start a workspace in prev env
- Wait until the workspace is ready
- Connect with ssh copy-paste, get `conn A`
    - Failed with ssh key conn
    - Success with access token conn
- add a public key via setting -> ssh keys
- Connect with ssh copy-paste use `ssh key`, get `conn B`
- Delete that public key you set in the settings
- Connection B should still work
- A new `conn C` which use the same command like conn B, will not works - ask a password


## Release Notes

```release-note
Allow users to add and remove SSH public keys in settings
Change UX of SSH copy-paste to support ssh key connection
```

## Documentation


## Werft options:
